src/tracks/PatternTrack.cpp

```diff
@@ -113,8 +115,23 @@ bool PatternTrack::play( const TimePos & _start, const fpp_t _frames,
 	{
 		if (!clip->isMuted() && clip->startPosition() >= lastPosition)
 		{
-			lastPosition = clip->startPosition();
-			lastLen = clip->length();
+			/**
+				calculate an offset in playback according to the start time offset of the clip
+			*/
+			bar_t patLen = Engine::patternStore()->lengthOfPattern(dynamic_cast<PatternClip*>(clip)->patternIndex());
+			tick_t patTicks = patLen * TimePos::ticksPerBar();
+			tick_t offset = fmod(static_cast<double>(clip->startTimeOffset()) / TimePos::ticksPerBar(), patLen)
+					* TimePos::ticksPerBar();
+			if (offset == 0)
+			{
+				lastPosition = clip->startPosition();
+				lastLen = clip->length();
+			}
+			else
+			{
+				lastPosition = clip->startPosition() - (patTicks - offset);
```

Comment:
This will affect the condition at the beginning of the loop that attempts to find the clip with the latest start time. I'm not sure how much that matters, but perhaps it would better to avoid it by storing `patTicks - offset` in a `lastOffset` variable or something, then adding that in the call to `play` below?

Reply:
Now as you say, I am astonished it worked before, lol. It should be more self-explaining now. I also reduced the logic a little.